lumicks/pylake/nb_widgets/kymotracker_widgets.py

```diff
@@ -516,6 +521,36 @@ def __init__(
             "sigma_cutoff": sigma_cutoff,
         }
 
+        position_scale = calibrated_kymo_channel._pixel_size
+        vel_calibration = position_scale / calibrated_kymo_channel.line_time_seconds
+
+        self._slider_ranges = {
+            "window": (1, 15),
+            "pixel_threshold": (1, np.max(calibrated_kymo_channel.data)),
+            "line_width": (0.0, 15.0 * position_scale),
+            "sigma": (1.0 * position_scale, 5.0 * position_scale),
+            "vel": (-5.0 * vel_calibration, 5.0 * vel_calibration),
+        }
+        for key, slider_range in slider_ranges.items():
+            if key not in self._slider_ranges:
+                raise KeyError(
+                    f"Slider range provided for parameter that does not exist ({key}) "
+                    f"Valid parameters are: {list(self._slider_ranges.keys())}"
+                )
+
+            if len(slider_range) != 2:
+                raise ValueError(
+                    f"Slider range for parameter {key} should be given as "
+                    f"(lower bound, upper bound)."
+                )
+
+            if slider_range[1] < slider_range[0]:
+                raise ValueError(
+                    f"Lower bound should be lower than upper bound for parameter {key}"
+                )
+
+            self._slider_ranges[key] = slider_range
+
```

Comment:
Maybe add a `raise ValueError` for giving a range of negative values for the line width, line length, pixel threshold, window and sigma.

Reply:
Added an exception for it.

Reply:
I'd drop it, for being too strict. The current class of parameters might only make sense as positive integral types, but what if new parameters were to be added? Would this restriction automatically apply to those as well?

Technically, you'd have to specify the concrete type of each parameter, if you wanted to do it correctly.

Also, what's the harm of _allowing_ negative slider values for those parameters? Sure, you'd have buggy sliders, but those are still user input to the algorithm itself (the sliders only offer convenience to enter those parameters), and it's the responsibility of the algorithm to check the input and the parameters, IMO.

Reply:
It does make sense that the algorithm should check for the validity of the input, not the sliders.

(Btw, in the current version the velocity is a parameter that can be negative).

Reply:
The list of which parameters should be non-negative was explicitly set in the last commit though? Velocity wasn't part of it.
```non_negative = ["window", "pixel_threshold", "line_width", "sigma", "min_length"]```

I can drop the last commit (and not enforce positivity). Currently, the downstream algorithm does not enforce positivity. I can make a ticket for that, but I wouldn't piggyback that on this PR.

Personally, I don't feel strongly about it either way. The reason I checked for increasing bounds is because otherwise it breaks _before_ it gets to the slider (it's a widget requirement) with a cryptic error message. Hence it felt like a requirement for the input to this function.

So final verdict, enforce positive sliders where applicable, yay or nay?

Reply:
I don't feel strongly about it either, my answer is 'may'.

You could also change it into a warning :P 

Btw, I mentioned the velocity, to make clear that not all of the parameters are currently nonnegative. I was responding to the sentence 'but what if new parameters were to be added? Would this restriction automatically apply to those as well?'

Reply:
Alright. I've dropped the positive constraint for now and made a ticket for handling this at the algorithm level in the future.